### ISSUE TITLE
Handling non-ascii characters

### ISSUE THREAD

**Issue by oriordan** [PriNova/CodyAgentPy#10]
**Describe the bug**
When sending a chat message with non-ascii characters, the cody agent binary raises an error.

**To Reproduce**
```bash
$ python main.py
--- Create Server Connection ---
--- Initialize Agent ---
--- Retrieve Chat Models ---
... print of models redacted ...
--- Create new chat ---
--- Set Model ---
--- Set context repo ---
--- Send message (short) ---
Human: tschüss, cody!
jsonrpc.ts: JSON parse error against input '{"jsonrpc":"2.0","id":7,"method":"chat/submitMessage","params":{"id":"a2ffa4d7-1d50-4f2e-83ec-49219c693ee9","message":{"command":"submit","text":"tschüss, cody!","submitType":"user","addEnhancedContext":false,"contextFiles":[{"type":"file","uri":{"fsPath":"/home/prinova/CodeProjects/CodyAgentPy/main.py","path":"/home/prinova/CodeProjects/CodyAgentPy/main.py"}},{"type":"file","uri":{"fsPath":"/home/prinova/CodeProjects/CodyAgentPy/codypy/logger.py","path":"/home/prinova/CodeProjects/CodyAgentPy/codypy/logger.py"}}]}}', contentLengthRemaining=0. Error:
SyntaxError: Unexpected end of JSON input
Traceback (most recent call last):
  File "/home/oriordan/github/codypy/main.py", line 118, in <module>
    asyncio.run(main())
  File "/opt/python/3.12/lib64/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/opt/python/3.12/lib64/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/python/3.12/lib64/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/oriordan/github/codypy/main.py", line 89, in main
    response, context_files_response = await cody_agent.chat(
                                       ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oriordan/github/codypy/codypy/agent.py", line 275, in chat
    result = await request_response(
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oriordan/github/codypy/codypy/messaging.py", line 271, in request_response
    async for response in _handle_server_respones(reader):
  File "/home/oriordan/github/codypy/codypy/messaging.py", line 92, in _handle_server_respones
    response: str = await _receive_jsonrpc_messages(reader)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/oriordan/github/codypy/codypy/messaging.py", line 61, in _receive_jsonrpc_messages
    headers: bytes = await asyncio.wait_for(reader.readuntil(b"\r\n\r\n"), timeout=5.0)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/python/3.12/lib64/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
  File "/opt/python/3.12/lib64/python3.12/asyncio/streams.py", line 657, in readuntil
    raise exceptions.IncompleteReadError(chunk, None)
asyncio.exceptions.IncompleteReadError: 0 bytes read on a total of undefined expected bytes
```

**Expected behavior**
Ideally, unicode messages would be accepted by the cody agent binary and carried around appropriately.

**Additional context**
All messages (sent and received) are unicode, yet there is some encoding problem in Cody agent.
Non-ascii responses are arriving correctly, so that part works well.

**Comment by oriordan** [comment]
Workaround: until this is fixed upstream, we can use `unidecode` to replace all non-ascii characters we send to the agent binary. This has been tested and works well.

EDIT: After some further investigation the issue turned out to be the message length discrepancy between unicode chars vs. same string as bytes.

**Comment by PriNova** [comment]
issue fixed #11 

Thank you @oriordan
